src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx

```diff
@@ -0,0 +1,225 @@
+	Icon,
+	IconName,
+	MenuContent,
+	TextInput,
+} from '@viaa/avo2-components';
+import { isEmpty, isNil, truncate } from 'lodash';
+import React, { FC, useState } from 'react';
+
+import withUser, { UserProps } from '../../hocs/withUser';
+import useTranslation from '../../hooks/useTranslation';
+
+import { shareUserRightToString, sortShareUsers } from './ShareWithColleagues.helpers';
+import './ShareWithColleagues.scss';
+import { ShareUserInfo, ShareUserInfoRights } from './ShareWithColleagues.types';
+
+type ShareWithColleaguesProps = {
+	users: ShareUserInfo[];
+	onAddNewUser: (info: Partial<ShareUserInfo>) => void;
+	onEditRights: (info: ShareUserInfo, newRights: ShareUserInfoRights) => void;
+	onDeleteUser: (info: ShareUserInfo) => void;
+};
+
+const ShareWithColleagues: FC<ShareWithColleaguesProps & UserProps> = ({
+	users,
+	user,
+	onAddNewUser,
+	onDeleteUser,
+	onEditRights,
+}) => {
+	const { tText } = useTranslation();
+	const currentUser = users.find((u) => u.email === user?.mail) as ShareUserInfo;
+	const [isRightsDropdownOpen, setIsRightsDropdownOpen] = useState<boolean>(false);
+	const [newShareUser, setNewShareUser] = useState<Partial<ShareUserInfo>>({
+		email: undefined,
+		rights: undefined,
+	});
+	const [error, setError] = useState<string | null>(null);
+
+	const handleRightsButtonClicked = () => {
+		setIsRightsDropdownOpen(!isRightsDropdownOpen);
+	};
+
+	const handleAddNewUser = () => {
+		if (/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]+$/.test(newShareUser.email as string)) {
+			onAddNewUser(newShareUser);
+			setError(null);
+		} else {
+			setError(
+				tText(
+					'shared/components/share-with-colleagues/share-with-colleagues___email-is-geen-geldig-emailadres'
+				)
+			);
+		}
+	};
+
+	const handleEditUserRights = (info: ShareUserInfo, newRights: ShareUserInfoRights) => {
+		onEditRights(info, newRights);
+	};
+
+	const handleDeleteUser = (info: ShareUserInfo) => {
+		onDeleteUser(info);
+	};
+
+	const updateNewShareUserInfo = (value: Record<string, string>) => {
+		setNewShareUser({
+			...newShareUser,
+			...value,
+		});
+	};
+
+	const renderColleaguesInfoList = () => {
+		if (users.length > 0) {
+			return (
+				<ul className="c-colleagues-info-list">
+					{sortShareUsers(users).map((user, index) => (
+						<li key={index} className="c-colleague-info-row">
+							<div className="c-colleague-info-row__avatar">
+								<Avatar
+									initials={
+										user.firstName && user.lastName
+											? user.firstName[0] + user.lastName[0]
+											: user.email.slice(0, 2).toUpperCase()
+									}
+									image={user.profileImage}
+								/>
+							</div>
+
+							<div className="c-colleague-info-row__info">
+								{(user.firstName || user.lastName) && (
+									<p>{`${user.firstName} ${user.lastName}`}</p>
+								)}
+
+								<p className="c-colleague-info-row__info__email">
+									{truncate(user.email, {
+										length: 32,
+										separator: ' ',
+									})}
```

Comment:
wouldn't this be better truncated with css: 
https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow